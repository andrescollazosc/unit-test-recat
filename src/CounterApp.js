import React, { useState } from "react";
import PropTypes from "prop-types";

const CounterApp = ({ value = 100 }) => {
  const [counter, setCounter] = useState(value);

  // HandleAdd
  const handleAdd = (e) => {
    setCounter(counter + 1);
  };

  const resetValue = () => {
    setCounter(value);
  };

  const handleSubstract = (e) => {
    setCounter(counter - 1);
  };

  return (
    <>
      <h1>CounterApp</h1>
      <h2>{counter}</h2>

      <button onClick={handleAdd}>+1</button>
      <button onClick={resetValue}>Reset</button>
      <button onClick={handleSubstract}>-1</button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number,
};

export default CounterApp;
