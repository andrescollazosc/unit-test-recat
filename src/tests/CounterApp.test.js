import { shallow } from "enzyme";
import React from "react";
import CounterApp from "../CounterApp";

describe("Test <CounterApp />", () => {
  let wrapper = shallow(<CounterApp />);

  beforeEach(() => {
    wrapper = shallow(<CounterApp />);
  });

  test("should be show CounterApp succesfuly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Must be show default value = 100", () => {
    const wrapper = shallow(<CounterApp value={100} />);

    const counterText = wrapper.find("h2").text().trim();

    expect(counterText).toBe("100");
  });

  test("should increment with button plus 1", () => {
    wrapper.find("button").at(0).simulate("click");
    const counterText = wrapper.find("h2").text().trim();

    expect(counterText).toBe("101");
  });

  test("should decrement with button minus 1", () => {
    wrapper.find("button").at(2).simulate("click");
    const counterText = wrapper.find("h2").text().trim();

    expect(counterText).toBe("99");
  });
});

describe("When button reset is click", () => {
  test("should reset value in 100", () => {
    const wrapper = shallow(<CounterApp value={15}/>);
    wrapper.find("button").at(1).simulate("click");
    const counterText = wrapper.find("h2").text().trim();

    expect(counterText).toBe("15");
  });
});
