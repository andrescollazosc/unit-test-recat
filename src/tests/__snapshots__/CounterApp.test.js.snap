// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <CounterApp /> Must be show default value = 100 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    100
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;

exports[`Test <CounterApp /> should be show CounterApp succesfuly 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    100
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
